ivadomed/loader/files_dataset.py

```diff
@@ -0,0 +1,239 @@
+from __future__ import annotations
```

Comment:
Yeah, partially. 

So `files_dataset` has two major part, one part ,is the `files_pairs`, the other part is the config: `GeneralizedLoaderConfiguration`. 

The 2nd part ties into your earlier questioon on why that class is used (mostly when we construct new FIlesDataset) and contain information to make it compatible with MRI2DSegmentationDatset.

Reply:
Thanks for the clarification. It took me a while but I think I get it now.
It is called `FilesDataset` because it serves the same purpose as the current (and future) `BidsDataset` and we would eventually have a `RegexDataset` as well, is that correct?

Reply:
yeah, `FilesDataset` is like... the current v1 `BidsDataset` but without `BIDS`. I am still debating if `RegexDataset` needs to be a separate class or based on existing `FilesDataset` implementation. BidsDataset V2 will likely be a rewrite but leverage some similar high level logic from `FilesDataset`... to be finalized.